staticcheck/lint.go

```diff
@@ -117,6 +117,41 @@ func checkValidHostPort(arg int) CallCheck {
 	}
 }
 
+func checkNonOverlappingDstSrc(dstArg, srcArg int) CallCheck {
+	return func(call *Call) {
+		dst := call.Args[dstArg]
+		src := call.Args[srcArg]
+		_, dstConst := irutil.Flatten(dst.Value.Value).(*ir.Const)
+		_, srcConst := irutil.Flatten(src.Value.Value).(*ir.Const)
+		if dstConst || srcConst {
+			// one of the arguments is nil, therefore overlap is not possible
+			return
+		}
+		if dst.Value == src.Value {
+			// simple case of f(b, b)
+			dst.Invalid("overlapping dst and src")
+			return
+		}
+		dstSlice, ok := irutil.Flatten(dst.Value.Value).(*ir.Slice)
+		if !ok {
```

Comment:
For the future (because it's not applicable here once you fix the panics): you could drop the `, ok` and just let Go panic on the type assertion.